connect-file-pulse-filters/src/main/java/io/streamthoughts/kafka/connect/filepulse/filter/NamingConventionFilter.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 StreamThoughts.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.streamthoughts.kafka.connect.filepulse.filter;
+
+import static io.streamthoughts.kafka.connect.filepulse.config.NamingConvention.namingConventionNotFoundErrorMsg;
+
+import io.streamthoughts.kafka.connect.filepulse.config.NamingConvention;
+import io.streamthoughts.kafka.connect.filepulse.config.NamingConventionFilterConfig;
+import io.streamthoughts.kafka.connect.filepulse.data.TypedField;
+import io.streamthoughts.kafka.connect.filepulse.data.TypedStruct;
+import io.streamthoughts.kafka.connect.filepulse.reader.RecordsIterable;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.StreamSupport;
+import org.apache.commons.text.CaseUtils;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+
+public class NamingConventionFilter extends AbstractRecordFilter<NamingConventionFilter> {
```

Comment:
I still think this name does not correctly capture what the filter is doing, let's have a discussion offline on this